File: emotion-back/content/src/main/java/com/example/content/mapper/NoteMapper.java
package com.example.content.mapper;

import com.example.content.dtos.note.NoteCreateDto;
import com.example.content.dtos.note.NoteResponseDto;
import com.example.content.dtos.note.NoteShortDto;
import com.example.content.dtos.note.NoteUpdateDto;
import com.example.content.entities.Note;
import org.mapstruct.*;

@Mapper(componentModel = "spring", unmappedTargetPolicy = ReportingPolicy.IGNORE)
public interface NoteMapper {
    @Mapping(target = "id", ignore = true)
    @Mapping(target = "album", ignore = true)
    @Mapping(target = "createdAt", ignore = true)
    Note toEntity(NoteCreateDto dto);

    @Mapping(target = "albumId", source = "album.id")
    NoteResponseDto toDto(Note note);

    @Mapping(target = "albumId", source = "album.id")
    @Mapping(target = "albumTitle", source = "album.title")
    NoteShortDto toShortDto(Note note);

    @BeanMapping(nullValuePropertyMappingStrategy = NullValuePropertyMappingStrategy.IGNORE)
    @Mapping(target = "id", ignore = true)
    @Mapping(target = "album", ignore = true)
    @Mapping(target = "createdAt", ignore = true)
    void updateEntityFromDto(NoteUpdateDto dto, @MappingTarget Note note);
}

File: Android/app/src/main/java/com/example/emotionapp/data/mappers/NoteMapper.kt
package com.example.emotionapp.data.mappers

import com.example.emotionapp.data.dtos.NoteResponseDTO
import com.example.emotionapp.data.dtos.NoteShortDTO
import com.example.emotionapp.domain.entities.Note
import com.example.emotionapp.domain.entities.NoteSummary

fun NoteResponseDTO.toDomain(): Note = Note(
    id = id,
    albumId = albumId,
    title = title,
    content = content,
    createdAt = createdAt
)

fun Note.toDTO(): NoteResponseDTO = NoteResponseDTO(
    id = id,
    albumId = albumId,
    title = title,
    content = content,
    createdAt = createdAt
)

fun NoteShortDTO.toDomain(): NoteSummary = NoteSummary(
    id = id,
    title = title,
    createdAt = createdAt
)
